aggregate/aggregate

```diff
@@ -0,0 +1,189 @@
+    mentha_df["Interactor"] = mentha_df["Interactor"].str.replace(r"[ {].*", "", regex=True)
+
+    # Initialize the "Structure" column with the PDB id values:
+    mentha_df["Structure"] = mentha_df["PDB id"].fillna("")
+
+    # Add "AF_Huri_HuMAP" to "Structure" if HuMap or HuRI values are not "":
+    mentha_df["Structure"] = mentha_df.apply(
+        lambda row: ",".join(filter(None, [row["Structure"], "AF_Huri_HuMAP"])) \
+        if row["pDockQ HuMap"] != "" or row["pDockQ HuRI"] != "" else row["Structure"],
+        axis=1
+    )
+
+    # Drop columns "PDB id", "pDockQ HuMap", and "pDockQ HuRI":
+    mentha_df.drop(columns=["PDB id", "pDockQ HuMap", "pDockQ HuRI"], inplace=True)
+
+    # Group by all columns except "Structure" and aggregate "Structure" as comma-separated strings
+    mentha_df = mentha_df.groupby([
+        "Target_Uniprot_AC", 
+        "Target_protein", 
+        "Interactor_UniProt_AC", 
+        "Interactor", 
+        "Mentha_score"
+    ], as_index=False).agg({"Structure": lambda x: ",".join(sorted(set(filter(None, x))))})
+
+    mentha_df.sort_values(by="Mentha_score", ascending=False, inplace=True)
+
+    ## STRING2PDB OUTPUT PRE-PROCESSING: ##
+
+    string_df.fillna("", inplace=True)
+
+    # Keep only required columns from string_df:
+    string_df = string_df[[
+        "Target_protein", 
+        "Target_Uniprot_AC", 
+        "Interactor", 
+        "Interactor_UniProt_AC", 
+        "String_score", 
+        "PDB_ID"
+    ]]
+
+    # Group by relevant columns and aggregate PDB_IDs:
+    string_df = string_df.groupby([
+        "Target_protein", 
+        "Target_Uniprot_AC", 
+        "Interactor", 
+        "Interactor_UniProt_AC", 
+        "String_score"
+    ], as_index=False).agg({
+    "PDB_ID": lambda x: ",".join(sorted(set(map(str, filter(None, x)))))
+    })
+
+    string_df.sort_values(by="String_score", ascending=False, inplace=True)
+
+    if mentha_df.empty and string_df.empty:
+
+        print(f"Warning:No results found in Mentha and STRING for the query protein")
+    
+    if not string_df.empty and not mentha_df.empty:
+
+        mentha_target_ac = mentha_df["Target_Uniprot_AC"].iloc[0]
+        string_target_ac = string_df["Target_Uniprot_AC"].iloc[0]
+        if mentha_target_ac != string_target_ac:
+            print(f"Error: Target Uniprot ACs do not match between Mentha and STRING outputs.")
+            exit(1)
+
+    ## MERGING ##
+    # Merge mentha_df and string_df:
+    merged_df = pd.merge(
+        mentha_df, 
+        string_df[["Interactor_UniProt_AC", "String_score", "PDB_ID"]], 
+        on="Interactor_UniProt_AC", 
+        how="left"
+    )
+
+    # Update Structure column:
+    merged_df["Structure"] = merged_df.apply(
+        lambda row: ",".join(filter(None, sorted(set(row["Structure"].split(",") + row["PDB_ID"].split(",")))))
+        if pd.notna(row["PDB_ID"]) else row["Structure"],
```

Comment:
can you add an indentation level here? So it's more clear that the this line is part of the lambda function definition